dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datafile/FileServiceIT.java

```diff
@@ -0,0 +1,213 @@
+package edu.harvard.iq.dataverse.datafile;
+
+import edu.harvard.iq.dataverse.DatasetDao;
+import edu.harvard.iq.dataverse.DataverseRequestServiceBean;
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.EjbDataverseEngine;
+import edu.harvard.iq.dataverse.arquillian.arquillianexamples.WebappArquillianDeployment;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.engine.command.impl.PublishDatasetCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.PublishDataverseCommand;
+import edu.harvard.iq.dataverse.persistence.datafile.DataFile;
+import edu.harvard.iq.dataverse.persistence.datafile.FileMetadata;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion.VersionState;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ejb.EJB;
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.everyItem;
+import static org.hamcrest.Matchers.in;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class FileServiceIT extends WebappArquillianDeployment {
+    private static final long DRAFT_DATASET_WITH_FILES_ID = 52;
+
+    @EJB
+    private FileService fileService;
+
+    @EJB
+    private EjbDataverseEngine commandEngine;
+
+    @EJB
+    private AuthenticationServiceBean authenticationServiceBean;
+
+    @EJB
+    private DatasetDao datasetDao;
+
+    @Inject
+    private DataverseSession dataverseSession;
+
+    @Inject
+    private DataverseRequestServiceBean requestService;
+
+    // -------------------- TESTS --------------------
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void deleteFile_forDraftDataset() {
```

Comment:
It would be nice if we would test if file is actually gone from the filesystem. It will require to store some file in predefined location. Currently service will silently skip this step (file is not in there - so there is no problem)